src/shared/components/timer/timer.component.js

```diff
@@ -0,0 +1,123 @@
+import React, { PureComponent } from 'react';
+import moment from 'moment';
+import firebase from 'firebase';
+
+import { StartButton, SaveButton, Time, HistoryContainer } from './timer.styles';
+
+require('firebase/firestore');
+
+export class Timer extends PureComponent {
+  state = {
+    counting: false,
+    time: 0,
+    databaseTimes: [],
+  };
+
+  componentDidMount() {
+    this.db = firebase.firestore();
+    const times = [];
+    this.db
+      .collection('times')
+      .get()
+      .then(snapshot => {
+        snapshot.docs.map(doc => {
```

Comment:
`.map` zwraca nowy array, więc każde wywołanie callbacku powinno zwrócić wartość. Poza tym powinieneś unikać w nim side-effectów.

Użyj `forEach` lub `for`. Jeżeli zakładasz, że danych będzie mało to wystarczy `forEach`. Jeżeli będzie ich więcej to lepszym rozwiązaniem jest `for` (jest szybszy).